src/main/kotlin/org/pkl/lsp/resolvers/ResolveVisitors.kt

```diff
@@ -426,14 +426,19 @@ object ResolveVisitors {
         get() = expectedName
     }
 
-  fun completionItems(base: PklBaseModule): ResolveVisitor<List<CompletionItem>> {
-    return object : ResolveVisitor<List<CompletionItem>> {
+  fun completionItems(
+    base: PklBaseModule,
+    prefix: String? = null,
+  ): ResolveVisitor<Set<CompletionItem>> {
+    return object : ResolveVisitor<Set<CompletionItem>> {
       override fun visit(
         name: String,
         element: PklNode,
         bindings: TypeParameterBindings,
         context: PklProject?,
       ): Boolean {
+        if (prefix != null && !name.startsWith(prefix)) return true
```

Comment:
Probably better to just send over all completions and let the IDE present each of these options.

The editor should use this to let users fuzzy-find completion members (and even matches on names with mismatched cases)

![Screenshot 2024-09-24 at 7 21 54 AM](https://github.com/user-attachments/assets/e1627b23-a381-4f17-8e31-e7f9f1fb21b8)


```suggestion
```

Reply:
You are right, I somehow missed that the client would do the filtering.